src/main/java/com/gregtechceu/gtceu/integration/jade/provider/RecipeOutputProvider.java

```diff
@@ -110,28 +118,33 @@ protected void write(CompoundTag data, RecipeLogic recipeLogic) {
     protected void addTooltip(CompoundTag capData, ITooltip tooltip, Player player, BlockAccessor block,
                               BlockEntity blockEntity, IPluginConfig config) {
         if (capData.getBoolean("Working")) {
-            List<ItemStack> outputItems = new ArrayList<>();
+            List<Ingredient> outputItems = new ArrayList<>();
             if (capData.contains("OutputItems", Tag.TAG_LIST)) {
                 ListTag itemTags = capData.getList("OutputItems", Tag.TAG_COMPOUND);
                 if (!itemTags.isEmpty()) {
                     for (Tag tag : itemTags) {
                         if (tag instanceof CompoundTag tCompoundTag) {
-                            var stack = GTUtil.loadItemStack(tCompoundTag);
-                            if (!stack.isEmpty()) {
-                                outputItems.add(stack);
+                            var ingredient = GTUtil.loadIngredient(tCompoundTag);
+                            if (!ingredient.isEmpty()) {
+                                outputItems.add(ingredient);
```

Comment:
here, do the same thing you do when getting the output fluids
```suggestion
                            if (tCompoundTag.contains("count_provider")) {
                                var ingredient = IntProviderIngredient.SERIALIZER.parse(NbtOps.INSTANCE.converTo(JsonOps.INSTANCE, tCompoundTag);
                                outputFluids.add(ingredient);
                            } else {
                                var stack = GTUtil.loadItemStack(tCompoundTag);
                                if (!stack.isEmpty()) {
                                    outputItems.add(SizedIngredient.create(stack));
                                }
                            }
```